src/base/bittorrent/session.h

```diff
@@ -219,6 +220,25 @@ namespace BitTorrent
         bool isDisableASMWhenCategorySavePathChanged() const;
         void setDisableASMWhenCategorySavePathChanged(bool value);
 
+        void setCategorySavePath(const QString &categoryName, const QString &value);
```

Comment:
I think you're completely right. 
I'll do that as soon as I have some time (likely this weekend). Just wanted to get this PR out there for comments (which have been very helpful).